src/targets/gpu/lowering.cpp

```diff
@@ -376,6 +377,41 @@ struct miopen_apply
             return mod->replace_instruction(ins, ins->get_operator(), inputs, ins->module_inputs());
         });
     }
+
+    /**
+     *  Adds reshape lazy to reshape ops that can be aliased instead of copied
+     */
+    void add_reshape_lazy_op()
+    {
+        apply_map.emplace("reshape", [=](instruction_ref ins) {
+            /* Attempt lazy reshape to allow for aliasing. Potentially throws in get_shape if unable
+             * to alias */
+            try
+            {
+                auto lazy_ins = mod->replace_instruction(
+                    ins,
+                    make_op("reshape_lazy",
+                            {{"dims", {ins->get_operator().to_value().at("dims")}}}),
+                    ins->inputs(),
+                    ins->module_inputs());
+                return lazy_ins;
+            }
+            catch(...)
```

Comment:
You shouldn't need the `catch`, but maybe you need #2038. Also the catch may leave the module in a bad state as well since you are modifying the module.

Reply:
Instead of a catch or using #2038, you can just always insert gpu::contiguous -> reshape_lazy -> gpu::contiguous. The eliminate_contiguous will take care of removing unnecessary contiguous later on.

Reply:
Noted. Removed the try/catch. Added contiguous to the end but not infront. Finding that adding the contiguous before the reshape_lazy causes failures with `gemm_2args_vv` test